android/java/org/chromium/chrome/browser/vpn/billing/InAppPurchaseWrapper.java

```diff
@@ -137,16 +138,11 @@ public void onBillingSetupFinished(@NonNull BillingResult billingResult) {
         }
     }
 
-    public boolean isSubscriptionSupported() {
+    public void endConnection() {
         if (mBillingClient != null) {
```

Comment:
we should call `endConnection` on `onBillingServiceDisconnected` as well before we try to reconnect. `Note: This does not remove the billing service connection itself - this binding to the service will remain active` https://developer.android.com/reference/com/android/billingclient/api/BillingClientStateListener#onBillingServiceDisconnected()